### PR TITLE
feat/add more workload types

### PR DESCRIPTION
**Pull request by jonathan-mayer**
## Motivation

Closing #20 and #56 and #60

## Changes

Add workload types:
- argo rollouts
- zalando stacks
- monitoring prometheuses

Refactor scaling for replica-based workloads to have less duplicate code

## Tests done

- [X] unit tests passed

### CODE REVIEW

File: internal/pkg/scalable/stacks.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	zalandov1 "github.com/zalando-incubator/stackset-controller/pkg/apis/zalando.org/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// getStacks is the getResourceFunc for Zalando Stacks
+func getStacks(namespace string, clientsets *Clientsets, ctx context.Context) ([]Workload, error) {
+	var results []Workload
+	stacks, err := clientsets.Zalando.ZalandoV1().Stacks(namespace).List(ctx, metav1.ListOptions{TimeoutSeconds: &timeout})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get deployments: %w", err)
```
Comment (JTaeuber) [inline internal/pkg/scalable/stacks.go:16]:
```suggestion
		return nil, fmt.Errorf("failed to get stacks: %w", err)
```

File: internal/pkg/scalable/stacks.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		return 0, errNoReplicasSpecified
+	}
+	return *s.Spec.Replicas, nil
+}
+
+// Update updates the resource with all changes made to it. It should only be called once on a resource
+func (s *stack) Update(clientsets *Clientsets, ctx context.Context) error {
+	_, err := clientsets.Zalando.ZalandoV1().Stacks(s.Namespace).Update(ctx, s.Stack, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update deployment: %w", err)
```
Comment (JTaeuber) [inline internal/pkg/scalable/stacks.go:48]:
```suggestion
		return fmt.Errorf("failed to update stack: %w", err)
```

File: internal/pkg/values/util.go [hunk 3]
```diff
@@ -96,14 +96,16 @@ func GetLayerFromAnnotations(annotations map[string]string, logEvent resourceLog
 		}
 	}
 	if downscaleReplicas, ok := annotations[annotationDownscaleReplicas]; ok {
```
Comment (JTaeuber) [inline internal/pkg/values/util.go:98]:
```suggestion
	if _, ok := annotations[annotationDownscaleReplicas]; ok {
```

**Review by JTaeuber — CHANGES_REQUESTED** [review]

File: internal/pkg/scalable/helpers_test.go
Comment (JTaeuber) [inline internal/pkg/scalable/helpers_test.go:1]: Do we use this file for other tests?

**Review by JTaeuber — CHANGES_REQUESTED** [review]

File: internal/pkg/scalable/suspendScaledWorkloads.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	suspendScaledResource
+}
+
+// ScaleUp scales up the underlying suspendScaledResource
+func (r *suspendScaledWorkload) ScaleUp() error {
+	r.setSuspend(false)
+	return nil
+}
+
+// ScaleUp scales down the underlying suspendScaledResource
```
Comment (JTaeuber) [inline internal/pkg/scalable/suspendScaledWorkloads.go:27]:
```suggestion
// ScaleDown scales down the underlying suspendScaledResource
```

**Review by JTaeuber — CHANGES_REQUESTED** [review]

File: internal/pkg/values/util.go (hunk 3 above)
Comment (jonathan-mayer) [reply internal/pkg/values/util.go:98]: Am I missing something here, the value is being used as an argument in strconv.ParseInt

File: internal/pkg/scalable/helpers_test.go
Comment (jonathan-mayer) [reply internal/pkg/scalable/helpers_test.go:1]: Its for helper functions for the tests

File: deployments/chart/Chart.yaml [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
 description: A Helm chart for deploying the go-kube-downscaler
-
 type: application
+keywords:
+  - kube-downscaler
+  - go
+  - downscaling
 version: 1.1.0
 appVersion: 1.1.0
+icon: https://github.com/caas-team/GoKubeDownscaler/blob/main/deployments/chart/icon.svg
```
Comment (jonathan-mayer) [inline deployments/chart/Chart.yaml:11]:
``` suggestion
icon: https://raw.githubusercontent.com/caas-team/GoKubeDownscaler/refs/heads/main/deployments/chart/icon.svg
```

**Review by JTaeuber — APPROVED** [review]
LGTM

**Pull request merged**